vendor/engines/bulk_email/app/models/bulk_email/delivery_form.rb

```diff
@@ -37,6 +39,20 @@ def deliver_all
 
     private
 
+    def custom_subject_present_if_required
+      return if product&.offline?
+      return if custom_subject.present?
+
+      errors.add(:custom_subject, "can't be blank")
+    end
+
+    def custom_message_present_if_required
+      return if product&.offline?
+      return if custom_message.present?
+
+      errors.add(:custom_message, "can't be blank")
```

Comment:
I think this can be `errors.add(:custom_message, :blank)` and it'll take the default I18n of "can't be blank".